colossalai/zero/sharded_optim/sharded_optim_v2.py

```diff
@@ -161,7 +161,8 @@ def step(self, *args, **kwargs):
                 # Since p.data is fp32 and p.col_attr.sharded_data_tensor is fp16
 
                 # TODO() optimize this line CPU (fp32) -> GPU (fp16)
-                colo_model_data_tensor_move(p, p.col_attr.sharded_data_tensor)
+                p.col_attr.sharded_data_tensor.reset_payload(
+                    colo_model_tensor_clone(p.half(), torch.cuda.current_device()))
```

Comment:
why not use colo_model_data_tensor_move?
```
colo_model_data_tensor_move(p.half().cuda(current_device(), p.col_attr.sharded_data_tensor)
```

Reply:
1. When reuse fp16 shard and offload grad, `sharded_data_tensor` is on CPU. `colo_model_data_tensor_move` calls `tensor.copy_()` which won't change the device of `sharded_data_tensor` to CUDA.
2. According to the profile result of @1SAA , `tensor.copy_()` is slower than `tensor.half().to(torch.cuda.current_device())`. Besides, the latter one is easy to profile using our `PCIEProfiler`.

Reply:
OK